File: removable/qa/p_qa_form.erl
-module(p_qa_form).

-compile(export_all).

-include("sticky.hrl").
-include("qa.hrl").

render(Method, Action, Qa, Tags) ->
    Js = ale:ff("p_qa_form.js"),
    ale:app_add_js(Js),

    Cancel = case Method of
        put -> [{a, [{href, ale:path(qa, show, [Qa#qa.id])}], ?T("Cancel")}, " "];
        _   -> ""
    end,
    {form, [{id, qa_form}, {method, post}, {action, Action}], [
        {input, [{type, hidden}, {name, "_method"}, {value, Method}]},

        {span, [{class, label}], ?T("Question")},
        {input, [{type, text}, {class, textbox}, {name, question}, {value, Qa#qa.question}]},

        {span, [{class, label}], ?T("Detail")},
        {textarea, [{name, detail}], Qa#qa.detail},

        h_tag:render_tag_selection(Tags),

        h_application:render_mathcha(),

        {input, [{type, submit}, {class, button}, {value, ?T("Save")}]}, " ", Cancel
    ]}.

File: removable/qa/p_qa_form.js
$('#qa_form input[type=submit]').click(function() {
    $(this).hide();
    $(this).after('<img class="ajax-loader" src="/static/img/spinner.gif" />');

    tinyMCE.triggerSave();

    var action   = $('#qa_form').attr('action');
    var _method  = $('#qa_form input[name=_method]').val();
    var question = $('#qa_form input[name=question]').val();
    var detail   = $('#qa_form textarea[name=detail]').val();
    var tags     = $('#qa_form input[name=tags]').val();
    var answer   = $('#qa_form input[name=answer]').val();
    var eAnswer  = $('#qa_form input[name=encrypted_answer]').val();
    var postData = {_method: _method, question: question, detail: detail, tags: tags, answer: answer, encrypted_answer: eAnswer};

    $.post(action, postData, function(data) {
        if (data.error) {
            alert(data.error);
            $('#qa_form .ajax-loader').remove();
            $('#qa_form input[type=submit]').show();
        } else {
            window.location.href = "/qas/" + data.atomic;
        }
    }, "json");
    return false;
});
